app/app.go

```diff
@@ -1781,3 +1788,29 @@ func (d *Deployments) search(
 		return d.inventoryClient.Search(ctx, tid, parms)
 	}
 }
+
+func (d *Deployments) UpdateDeploymentsWithArtifactName(
+	ctx context.Context,
+	artifactName string,
+) error {
+	// first check if there are pending deployments with given artifact name
+	exists, err := d.db.ExistUnfinishedByArtifactName(ctx, artifactName)
+	if err != nil {
+		return errors.Wrap(err, "looking for deployments with given artifact name")
+	}
+	if !exists {
+		return nil
+	}
+
+	// Assign artifacts to the deployments with given artifact name
+	artifacts, err := d.db.ImagesByName(ctx, artifactName)
+	if err != nil {
+		return errors.Wrap(err, "Finding artifact with given name")
+	}
+
+	if len(artifacts) == 0 {
+		return ErrNoArtifact
+	}
```

Comment:
this is actually interesting; I've been asked to use `else` couple of times, but I can't get a habit of using it;
I think it's mainly because I've been learned to not use it back in a days when I was learning C in my first job;
I'm not an expert, and I'm not a fan of such discussions, but there are some patterns and discussions around usage of `else` statement e.g. it's part of the [return early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8);
and I'm also just sharing :) (really)